control-plane/agents/src/bin/ha/cluster/server.rs

```diff
@@ -51,17 +64,47 @@ impl ClusterAgentOperations for ClusterAgentSvc {
             ));
         }
 
+        let ep: SocketAddr = request.endpoint().parse().unwrap();
```

Comment:
should handle error here, but also begs the question, should endpoint return a `SocketAddr` ?